### PR TITLE
[prometheus-adapter] allow using cluster roles for auth reader

### PR DESCRIPTION
**Pull request by vince-vibin**
#### What this PR does / why we need it
When deploying the Prometheus Adapter in a Namespace the `auth-reader` role is still created in the `kube-system` namespace. Deploying into `kube-system` is not wanted for a lot of enterprise clusters.
This give the option to deploy the `auth-reader` role as a cluster role rather then a normal role. Now the User can easily deploy everything into one namespace
#### Which issue this PR fixes

*(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*

- fixes #

#### Checklist

- [x] [DCO](https://github.com/prometheus-community/helm-charts/blob/main/CONTRIBUTING.md#sign-off-your-work) signed
- [x] Chart Version bumped
- [x] Title of the PR starts with chart name (e.g. `[prometheus-couchdb-exporter]`)

### CODE REVIEW

**Comment by vince-vibin** [comment]
Hi @hectorj2f, @MattiasGees, @steven-sheehy could you please check this out 🙂

File: charts/prometheus-adapter/Chart.yaml [hunk 1]
```diff
@@ -1,6 +1,6 @@
 apiVersion: v1
 name: prometheus-adapter
-version: 4.9.1
+version: 4.9.2
```
Comment (steven-sheehy) [inline charts/prometheus-adapter/Chart.yaml:3]: Should be a minor bump since it's a new feature.

```suggestion
version: 4.10.0
```

**Review by steven-sheehy — CHANGES_REQUESTED** [review]

File: charts/prometheus-adapter/Chart.yaml (hunk 1 above)
Comment (vince-vibin) [reply charts/prometheus-adapter/Chart.yaml]: should be fixed. forgot the DOC sign at first 😅

**Comment by mmcloud** [comment]
Hi, studying this PR makes me ask is more secure here? Deploying the adapter using a namespaced scoped role binding on the kube-system or in another namespace but using the cluster scoped resource?

**Pull request merged**